drivers/mpu9150/mpu9150.c

```diff
@@ -515,51 +509,51 @@ static int compass_init(mpu9150_t *dev)
     }
 
     /* Configure Power Down mode */
-    i2c_write_reg(DEV_I2C, DEV_COMP_ADDR, COMPASS_CNTL_REG, MPU9150_COMP_POWER_DOWN);
+    i2c_write_reg(DEV_I2C, DEV_COMP_ADDR, COMPASS_CNTL_REG, MPU9150_COMP_POWER_DOWN, 0);
```

Comment:
you miss the i2c_read_reg call 7 lines above

Reply:
oops, fixed!